### PR TITLE
Serde instance

### PR DESCRIPTION
**Pull request by AlexanderSaydakov**
deprecated class-level serde (to be removed later), support serde instance

### CODE REVIEW

**Review by jmalkin — CHANGES_REQUESTED** [review]
I also looked for any remaning instances of `S()` in the headers and the only instances aside from the new methods to handle the old calls was tuple sketch summary destructors.

I think the header size is not needed when serializing the gadget though? Let me know if that's incorrect and I can approve.

File: kll/include/kll_sketch_impl.hpp [hunk 1]
```diff
@@ -334,8 +334,8 @@ double kll_sketch<T, C, S, A>::get_normalized_rank_error(bool pmf) const {
 
 // implementation for fixed-size arithmetic types (integral and floating point)
 template<typename T, typename C, typename S, typename A>
-template<typename TT, typename std::enable_if<std::is_arithmetic<TT>::value, int>::type>
-size_t kll_sketch<T, C, S, A>::get_serialized_size_bytes() const {
+template<typename TT, typename SerDe, typename std::enable_if<std::is_arithmetic<TT>::value, int>::type>
+size_t kll_sketch<T, C, S, A>::get_serialized_size_bytes(const SerDe&) const {
```
[the remark this answers is not in the document]
Comment (AlexanderSaydakov) [reply kll/include/kll_sketch_impl.hpp:338]: the class-level parameter is only used as a default in the declaration

File: sampling/include/var_opt_union_impl.hpp [hunk 2]
```diff
@@ -279,7 +294,7 @@ std::vector<uint8_t, AllocU8<A>> var_opt_union<T,S,A>::serialize(unsigned header
     ptr += copy_to_mem(outer_tau_numer_, ptr);
     ptr += copy_to_mem(outer_tau_denom_, ptr);
 
-    auto gadget_bytes = gadget_.serialize();
+    auto gadget_bytes = gadget_.serialize(header_size_bytes, sd);
```
[the remark this answers is not in the document]
Comment (AlexanderSaydakov) [reply sampling/include/var_opt_union_impl.hpp]: fixed. thanks

**Review by jmalkin — APPROVED** [review]

**Comment by jmalkin** [comment]
👍

**Pull request merged**